module/Photo/src/Photo/Controller/AlbumController.php

```diff
@@ -1,3 +1,104 @@
 <?php
 
+namespace Photo\Controller;
 
+use Zend\Mvc\Controller\AbstractActionController;
+use Zend\View\Model\ViewModel;
+
+class AlbumController extends AbstractActionController
+{
+
+    public function indexAction()
+    {
+        $album_id = $this->params()->fromRoute('album_id');
+        $activepage = (int) $this->params()->fromRoute('page');
+        $album_service = $this->getAlbumService();
+        $photo_service = $this->getPhotoService();
+        $album = $album_service->getAlbum($album_id);
+        $config = $album_service->getConfig();
+        $lastpage = (int) floor(($album->getPhotoCount() + $album->getAlbumCount()) / $config['max_photos_page']);
+        if ($activepage > $lastpage) {
+            //TODO: throw some error, if this every occurs it's the user's fault
+        }
+
+        $albums = array();
+        $album_start = $activepage * $config['max_photos_page'];
+        //check if we need to display albums on this page:
+        if ($album_start < $album->getAlbumCount()) {
+            $albums = $album_service->getAlbums($album, $album_start, $config['max_photos_page']);
+        }
+        $albums = $photo_service->populateCoverPhotos($albums);
+        $photos = array();
+        $photo_count = $config['max_photos_page'] - count($albums);
+        //check if we need to display photos on this page:
+        if ($photo_count > 0) {
+            $photo_start = max($activepage * $config['max_photos_page'] - $album->getAlbumCount(), 0);
+            $photos = $this->getAlbumService()->getPhotos($album, $photo_start, $photo_count);
+        }
+
+        //we'll fix this ugly thing later vv
+        $basedir = str_replace("public", "", $config['upload_dir']);
```

Comment:
Dit wordt hier alleen gebruikt om de correcte path te returnen om foto's weer te geven. Niet voor form uploads. Zodra issue #53 is opgelost komt hier een betere implementatie voor.